src/Lib/ImageTransformInterface.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/**
+ * @category Proffer
+ * @package ImageTransformInterface.php
+ *
+ * @author David Yell <neon1024@gmail.com>
+ * @when 23/03/15
+ *
+ */
+
+namespace Proffer\Lib;
+
+use Cake\ORM\Table;
+
+interface ImageTransformInterface
+{
+
+    /**
+     * @param Table $table Instance of the current table
+     * @param ProfferPathInterface $path Instance of the current path class
+     */
+    public function __construct(Table $table, ProfferPathInterface $path);
```

Comment:
Yes this is intentional as the dependency is injected during the construction, rather than being passed into each class method. As each method should use these dependencies.